app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/TaskListController.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend.controllers
+import com.kenshoo.play.metrics.Metrics
+import javax.inject.Inject
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent}
+import uk.gov.hmrc.agentsubscriptionfrontend.config.AppConfig
+import uk.gov.hmrc.agentsubscriptionfrontend.service.{SessionStoreService, SubscriptionService}
+import uk.gov.hmrc.auth.core.AuthConnector
+import uk.gov.hmrc.agentsubscriptionfrontend.views.html
+
+import scala.concurrent.ExecutionContext
+
+class TaskListController @Inject()(
+  override val authConnector: AuthConnector,
+  continueUrlActions: ContinueUrlActions,
+  val sessionStoreService: SessionStoreService)(
+  implicit override implicit val appConfig: AppConfig,
+  metrics: Metrics,
+  override val messagesApi: MessagesApi,
+  val ec: ExecutionContext)
+    extends AgentSubscriptionBaseController(authConnector, continueUrlActions, appConfig) with SessionBehaviour {
+
+  def showTaskList: Action[AnyContent] = Action.async { implicit request =>
+    withSubscribingAgent { implicit agent =>
+      sessionStoreService.fetchAgentSession.map {
+        case Some(session) => Ok(html.task_list(session.identifyBusinessTaskComplete))
```

Comment:
having a valid session doesn't necessarily mean that the business details task is done , we need to check if valid business details exist in the session or not ,
or am I missing something ?

Reply:
there's a flag in the session that gets set to true at the end of the business details section (after you confirm the business). If the business task isn't done the task list will still be shown but with only the business details link clickable so they can complete this section.

Reply:
ah yes, i misread the above logic , i thought you are simply doing `.task_list(identifyBusinessTaskComplete)` but you are actually doing `.task_list(session.identifyBusinessTaskComplete)` 

but don't we need to check for the continueUrl here too as if someone manually hits the  /task-list url (from Trusts) then the task list page will be shown , right ?

Reply:
yeah thats true, ok!